components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java

```diff
@@ -947,4 +959,48 @@ public Properties getResourceProperties(String entryKey) {
         }
         return null;
     }
+
+    /**
+     * Populate file extension to media type mapping
+     *
+     * @return Map which contains file extension to media type mapping
+     */
+    private Map<String, String> createFileExtensionsMap() {
+
+        Map<String, String> extensionContentTypeMap = new HashMap<>();
+
+        extensionContentTypeMap.put(".xml", "application/xml");
+        extensionContentTypeMap.put(".js", "application/javascript");
+        extensionContentTypeMap.put(".css", "text/css");
+        extensionContentTypeMap.put(".html", "text/html");
+        extensionContentTypeMap.put(".sql", "text/plain");
+        extensionContentTypeMap.put(".xsd", "Schema");
+        extensionContentTypeMap.put(".xsl", "application/xsl+xml");
+        extensionContentTypeMap.put(".xslt", "application/xslt+xml");
+        extensionContentTypeMap.put(".dmc", "application/datamapper");
+        extensionContentTypeMap.put(".zip", "application/zip");
+        extensionContentTypeMap.put(".wsdl", "WSDL");
+
+        return extensionContentTypeMap;
+    }
+
+    /**
+     * Loopkup media-type for the relevant file from the mappings
+     *
+     * @param fileUrl File URL of the registry resource
+     * @return Media-type of the file
+     */
+    private String lookUpFileMediaType(String fileUrl) {
+        String extension = "";
+        int i = fileUrl.lastIndexOf('.');
+        if (i > 0) {
+            extension = fileUrl.substring(i);
+        }
+
+        String mediaType = "";
+        if (fileExtensionMediaTypeMap != null && fileExtensionMediaTypeMap.containsKey(extension)) {
```

Comment:
Can the map be null at this point? It's initialized in the init method

Reply:
A redundant check. This code is changed in the latest branch.